c/src/point_one/polaris/polaris.c

```diff
@@ -543,26 +554,40 @@ int Polaris_Work(PolarisContext_t* context) {
 #endif
 
   if (bytes_read < 0) {
+    // Did we hit a read timeout? This is normal behavior (e.g., brief loss of
+    // cell service) and is not considered an error. Most receivers can handle
+    // small gaps in corrections data. We do not close the socket here.
+#ifdef POLARIS_USE_TLS
+    int ssl_error = SSL_get_error(context->ssl, bytes_read);
+    if (ssl_error == SSL_ERROR_WANT_READ || ssl_error == SSL_ERROR_WANT_WRITE) {
+#else
     if (errno == EAGAIN || errno == ETIMEDOUT) {
+#endif
       P1_DebugPrint("Socket timed out.\n");
       return 0;
```

Comment:
This is a nit, but we should define a POLARIS_OK or similar for 0. It's odd to me that we mix defines and numerical numbers.

Reply:
or we should define POLARIS_SOCKET_TIMEOUT as 0 if you think that is a bit too misleading.

Reply:
I don't think so in this case. This function very specifically returns `The number of received bytes, or 0 if the timeout elapsed.`. The fact that `POLARIS_SUCCESS` is defined as 0 is sort of irrelevant, and in theory it could just as easily be a positive number, which would not work here.

Alternatively, if we want to change this function to return `POLARIS_TIMED_OUT` on timeout and _never_ return 0, we could do that?

Reply:
I think if we're gonna define something it should NOT be 0. I'm not sure that we need a separate `POLARIS_SOCKET_TIMEOUT` vs the existing `POLARIS_TIMED_OUT`, but I'm open to using `POLARIS_TIMED_OUT` here if you think that's preferable?

Reply:
Ok I made that change. It's better to be explicit. We now return `POLARIS_TIMED_OUT`, and `Polaris_Work()` should never return 0 under any circumstances.

Reply:
I agree that's the cleanest solution.